Lesson 2 recap loops (#62)
ReDI-School/intermediate-java: recap-loops/index.md

## Exercises

> ### Sum of an array.

Reviewer: ```suggestion
### Exercise 1: Sum of an array.
```

> Write a method that takes an array of integers and returns the sum of the elements of said array.

Reviewer: adding instructions on how to test it (+ introducing some engineering vocab) and simplified the language a bit to be more inclusive to english beginners
```suggestion
Write a method that takes an array of integers and returns the sum of the elements of that array. Test it by passing in a few `hardcoded` inputs and printing the output to the console. 
Examples:
`1,2,3,4] `-> 10
`[]` -> 0
`[1]` -> 1
```

> ### Average

Reviewer: ```suggestion
### Exercise 2: Average
```

> The average of a collection of numbers is the sum divided by the number of elements. Write a method that takes an array of

Reviewer: ```suggestion
The average of a list of numbers is the sum of those numbers divided by the number of elements. Write a method that takes an array of integers and returns the average. Make sure to reuse the sum method that you have written for Exercise 1.
Examples:
`[1,2,3,4]` -> 2.5
`[10,10,10]` -> 10
`[]` -> null
```

> integers and returns the average. Make sure to reuse the sum method that you have written above.

Reviewer: ```suggestion
```

Reviewer: added to the above one without the linebreak

> ### Count the `char`

Reviewer: ```suggestion
### Exercise 3: Count the `char`
```

> Write a method that takes a `String` and a `char` and returns the number of times you can find that character in that

Reviewer: ```suggestion
Write a method that takes a `String` and a `char` and returns the number of times you can find that character in that string. Examples:
`"Hello!", 'l'` -> 2
`"Banana", 'a'` -> 3
`"ReDI", 'a'` -> 0 
```

> string.

Reviewer: ```suggestion
```

Reviewer: moved to above

> ### FizzBuzz
> 
> Write a program that prints the numbers from 1 to 100. But for multiples of three print “Fizz” instead of the number and for the multiples of five print “Buzz”. For numbers which are multiples of both three and five print “FizzBuzz”.
> Share your code (also with each other) using git.

Reviewer: ```suggestion
```

This was their entrance exam when they applied for the course.

Author: Thanks, good to know :)

## Homework

### Graded: Simple Multiplication Table

This exercise is available in [Github classroom assignment here](https://classroom.github.com/a/wJEsXx7a)

> - Read a one number with Scanner

Reviewer: ```suggestion
- Read a single number with [Scanner](https://beginnersbook.com/2017/09/java-program-to-read-integer-value-from-the-standard-input/)
```

Reviewer: Otherwise we might end up having a bunch of people completely lost

Author: We have examples how to use scanner in this class, but good addition anyway, tks

> - Print the multiplication table of the number from 1 till 10

Reviewer: ```suggestion
- Print the multiplication table of the number from 1 until 10
```

2 x 10 = 20
```

### Graded: Dividable numbers

This exercise is available in [Github classroom assignment here](https://classroom.github.com/a/KOXYDafF)

> - Read two numbers with Scanner

Reviewer: This one's gonna confuse a bunch of them for sure, I didn't get it either until I saw the examples.
Maybe something like `Write a function that returns the largest element in an array of integer` might be more appropriate here?

Author: Yes, I agree that it is not the most intuitive. However I laid out already the skeleton in the classroom assignment so that it becomes more intuitive.  Let me see if I can improve a bit on the description anyway.
